internals/metrics/metrics.go

```diff
@@ -27,6 +26,17 @@ const (
 	TypeGaugeInt
 )
 
+func (mt MetricType) String() string {
+	switch mt {
+	case TypeCounterInt:
+		return "counter"
+	case TypeGaugeInt:
+		return "gauge"
+	default:
+		panic("invalid metric type")
```

Comment:
Nit: best to include the type number in the message:
```suggestion
		panic(fmt.Sprintf("invalid metric type %d", mt))
```